### PR TITLE
Expose EFS toggle #109

### LINKED ISSUE #109: Expose EFS toggle
The upstream blueprint module for EKS addons exposes [enable_aws_efs_csi_driver](https://github.com/ntwkninja/terraform-aws-eks-blueprints/blob/v4.21.1/modules/kubernetes-addons/README.md#input_enable_aws_efs_csi_driver) as an input.

It would be nice to be able to toggle it on in our module, [similar to EBS](https://github.com/defenseunicorns/iac/blob/v0.0.0-alpha.5/modules/eks/eks-addons.tf#L20).

### PR DESCRIPTION
**Pull request by JaseKoonce**
Adds EFS toggle for another storage option in addition to EBS. Currently setting the default for EFS_ENABLED to false.

### CODE REVIEW

File: modules/eks/main.tf [hunk 1]
```diff
@@ -78,3 +80,67 @@ resource "aws_iam_role" "auth_eks_role" {
...
+    },
+    {
+      type        = "egress"
+      from_port   = 2049
+      to_port     = 2049
+      protocol    = "tcp"
+      cidr_blocks = ["0.0.0.0/0"]
+
+    }
+  ]
```
Comment (ntwkninja) [inline modules/eks/main.tf:145]: these look like the default with exception of the cidr_blocks. Recommend we expose the cidr_blocks as a var and input the in the root module like [this](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/examples/stateful/main.tf#L374-L379)

**Review by ntwkninja — COMMENTED** [review]
recommending minor changes and also we need to add a test (recommend a brief discussion on the latter)

File: modules/eks/main.tf (hunk 1 above)
Comment (JaseKoonce) [reply modules/eks/main.tf]: I went ahead and adjusted the security groups to utilize the cidr blocks from the vpc module. Let me know if that doesn't fit what you were thinking.

Comment (ntwkninja) [reply modules/eks/main.tf]: @JaseKoonce lgtm

@RothAndrew @brianrexrode do we want to test optional components or opinionate storage (EFS vs. EBS)?

Comment (RothAndrew) [reply modules/eks/main.tf]: Speaking generally, I believe we should test any combination of features that we believe people will use.

“If the pipeline says everything looks good, that should be enough - it forces the focus on what ‘releasable’ means.” - Dave Farley

Comment (RothAndrew) [reply modules/eks/main.tf]: If the question is whether to offer a variety of features or "hard code" a set of opinionated ones, then I believe we should start with the smallest "decision space" possible that meets the use case, and expand incrementally as we gather feedback on what is needed.

Comment (ntwkninja) [reply modules/eks/main.tf]: found a decent [article](https://containerjournal.com/topics/container-networking/using-ebs-and-efs-as-persistent-volume-in-kubernetes/) breaking down pros and cons

IMO, both are needed so a toggle seems appropriate

Comment (ntwkninja) [reply modules/eks/main.tf]: @JaseKoonce can you add a test when you have time.

Comment (JaseKoonce) [reply modules/eks/main.tf]: @ntwkninja I went ahead and added enable_efs to fixtures.common.tfvars and set it to true. Test is running now.

**Review by ntwkninja — APPROVED** [review]
🚀

**Pull request merged**